File: solutions/csharp/log-levels/2/LogLevels.cs
using System;

static class LogLine
{
    public static string Message(string logLine) =>
        logLine.Split(':')[1].Trim();

    public static string LogLevel(string logLine)
    {
        var firstLetterIndex = logLine.IndexOf('[') + 1;
        var rightBracketIndex = logLine.IndexOf(']');
        return logLine[firstLetterIndex..rightBracketIndex].ToLower();
    }

    public static string Reformat(string logLine) =>
        $"{Message(logLine)} ({LogLevel(logLine)})";
}

File: solutions/fsharp/log-levels/2/LogLevels.fs
module LogLevels

open System.Text.RegularExpressions

let pattern = 
    Regex("\[(\w+)\]:\s+(.*\w)")

let message (logLine: string): string = 
    pattern.Match(logLine).Groups[2].Value

let logLevel(logLine: string): string = 
    pattern.Match(logLine).Groups[1].Value.ToLower()

let reformat(logLine: string): string =
    let lvl = logLevel logLine
    let msg = message logLine
    sprintf $"{msg} ({lvl})"